### PR TITLE
Feature/map search

### PR DESCRIPTION
**Pull request by agdt3**
## TODOs:
* Move CSS into stylesheet
* Add clear search/show all button
* Only show markers that can be reached from start position
* Inject Maps API key

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -528,6 +569,40 @@ def _filter_carpools_by_date(leave_time, return_time):
...
+        Carpool.from_longitude == from_longitude,
+        Carpool.to_latitude == to_latitude,
+        Carpool.to_longitude == to_longitude
+    )
+
+    return pools
+
+
+def _get_all_carpool_locations():
+    pools = Carpool.query.all()
```
Comment (iandees) [inline app.py]: We probably don't want to load *all* the carpool start/end locations onto the map. That could be a lot of things.

Comment (agdt3) [inline app.py]: Yeah, scaling is going to change this. One option is to get the user's zip code or State and load just for that bounding box.

File: app.py [hunk 2]
```diff
@@ -528,6 +569,40 @@ def _filter_carpools_by_date(leave_time, return_time):
     return pools
 
 
+def _filter_carpools_by_location(from_latitude,
+                                 from_longitude,
+                                 to_latitude,
+                                 to_longitude):
+
+    pools = Carpool.query.filter(
+        Carpool.from_latitude == from_latitude,
```
Comment (agdt3) [inline app.py]: I'm not sure I follow. There are no operations being performed on floats (avoiding the pitfalls of float arithmetic), so those values should stay the same (we can even make them string fields and only convert them in on the front-end). Right now, the user simply clicks on a marker and is presented with a series of other markers to click on. So long as no one decides to actually add or multiply lat/lng floats, the values should just match. Unless I'm missing an obvious point about lat/lng attributes changing for addresses over time! However, even in this case, we're only reading the values from google maps api once (when the driver enters their address). We then populate the map and do our filtering from our own stored values.

If you're implying that the user should be able to search in a better way than by clicking on a marker, I totally agree, but that would require some design work.

Apologies if I'm missing something utterly obvious.

**Comment by agdt3** [comment]
"Figuring out how to get the lat/lon and location name data into the carpool create form might be a better place to start." You can currently do this by using the address autocomplete fields when adding a new route.

**Pull request closed without merging**